app/facades/movie_facade.rb

```diff
@@ -1,53 +1,95 @@
 class MovieFacade
 
-    def self.top_movies_endpoint(page_num)
-        sort_by = 'vote_average.desc'
-        conn.get("3/discover/movie?api_key=#{movies_api_key}&#{language('en-US')}&sort_by=#{sort_by}&#{exclude_adult}&page=#{page_num}&vote_count.gte=300")
-    end
+# def self.top_movies_endpoint(page_num)
+#     sort_by = 'vote_average.desc'
+#     conn.get("3/discover/movie?api_key=#{movies_api_key}&#{language('en-US')}&sort_by=#{sort_by}&#{exclude_adult}&page=#{page_num}&vote_count.gte=300")
+# end
 
-    def self.keyword_search_endpoint(page_num, keywords)
-        conn.get("3/search/movie?api_key=#{movies_api_key}&#{language('en-US')}&#{exclude_adult}&page=#{page_num}&query=#{keywords}")
-    end
+# def self.keyword_search_endpoint(page_num, keywords)
+#     conn.get("3/search/movie?api_key=#{movies_api_key}&#{language('en-US')}&#{exclude_adult}&page=#{page_num}&query=#{keywords}")
+# end
+
+# def self.movie_show_endpoint(id)
+#     conn.get("3/movie/#{id}?api_key=#{movies_api_key}")
+# end
+
+# def self.movie_cast_endpoint(id)
+#     conn.get("3/movie/#{id}/credits?api_key=#{movies_api_key}")
+# end
 
-    def self.keyword_search(keywords)
-        movies1 = MovieService.keyword_search(1, keywords)[:results].map do |movie_data|
-            MovieCreator.new(movie_data)
-        end
-        movies2 = MovieService.keyword_search(2, keywords)[:results].map do |movie_data|
-            MovieCreator.new(movie_data)
-        end
-        (movies1.concat(movies2)).flatten
+# def self.movie_review_endpoint(id)
+#     conn.get("3/movie/#{id}/reviews?api_key=#{movies_api_key}&#{language('en-US')}")
+# end
+
+def self.keyword_search(keywords)
+  # if MovieService.keyword_search(1, keywords)[:status]
+    # movies1 = MovieService.keyword_search(1, keywords)[:results].map do |movie_data|
+    #   MovieDetails.new(movie_data)
+    # end
+  # else
+    movies1 = MovieService.keyword_search(1, keywords)[:results].map do |movie_data|
+      MovieDetails.new(movie_data)
+    end
+    movies2 = MovieService.keyword_search(2, keywords)[:results].map do |movie_data|
+      MovieDetails.new(movie_data)
     end
+  # end
+  (movies1.concat(movies2)).flatten
+end
 
-    def self.top_movies
-        movies1 = MovieService.top_movies(1)[:results].map do |movie_data|
-            MovieCreator.new(movie_data)
-        end
-        movies2 = MovieService.top_movies(2)[:results].map do |movie_data|
-            MovieCreator.new(movie_data)
-        end
-        (movies1.concat(movies2)).flatten
+  def self.top_movies
+    movies1 = MovieService.top_movies(1)[:results].map do |movie_data|
+      MovieDetails.new(movie_data)
+    end
+    movies2 = MovieService.top_movies(2)[:results].map do |movie_data|
+      MovieDetails.new(movie_data)
     end
+    (movies1.concat(movies2)).flatten
+  end
+
+  def self.get_movie_details(id)
```

Comment:
This new method brings together all the private methods below, so now in our controller, we only have a single instance variable. @movies contains movie data, cast and review data objects